drop-transfer/src/service.rs

```diff
@@ -200,10 +199,16 @@ impl Service {
         let started = state.validate_for_download(file_id)?;
 
         if started {
-            validate_dest_path(parent_dir)?;
+            validate_dest_path(parent_dir.as_ref())?;
+            state.file_events(file_id)?.pending(parent_dir).await;
 
             state
-                .start_download(&self.state.storage, file_id, parent_dir, &self.logger)
+                .start_download(
+                    &self.state.storage,
+                    file_id,
+                    parent_dir.as_ref(),
+                    &self.logger,
```

Comment:
Why can't we do `&parent_dir` and use `.as_ref()` instead?

Reply:
Because `parent_dir` is now `str` instead of `Path`, so we need to make use of the `AsRef` trait instead of a simple reference to satisfy the `start_download()` signature